FLASK_APP = app.py
FLASK_ENV = development
FLASK_DEBUG = 0
In folder /home/alex/PycharmProjects/ReinforcementTrader
/home/alex/miniconda3/envs/ReinforcementTrader/bin/python -m flask run 
DEBUG:asyncio:Using selector: EpollSelector
INFO:data_manager:Data Manager initialized with data directory: data
INFO:ib_integration:IB Integration initialized
INFO:risk_manager:Risk Manager initialized
INFO:trading_engine:Trading Engine initialized with device: cuda
INFO:trading_engine:Default configurations initialized
INFO:websocket_handler:WebSocket handler initialized
INFO:werkzeug:WARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.
 * Running on http://127.0.0.1:5000
INFO:werkzeug:Press CTRL+C to quit
 * Serving Flask app 'app.py'
 * Debug mode: off
INFO:websocket_handler:Client T_Qru6Ctf0gBIq1rAAAB connected from 127.0.0.1
INFO:websocket_handler:Background broadcast thread started
INFO:werkzeug:127.0.0.1 - - [11/Jul/2025 15:19:17] "GET /socket.io/?EIO=4&transport=polling&t=PVxaaZj HTTP/1.1" 200 -
INFO:websocket_handler:Client JEUGqqLUVPBt5TmLAAAD connected from 127.0.0.1
INFO:werkzeug:127.0.0.1 - - [11/Jul/2025 15:19:17] "POST /socket.io/?EIO=4&transport=polling&t=PVxaaZt&sid=ufcpv6MMAw54iD0HAAAC HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [11/Jul/2025 15:19:17] "GET /socket.io/?EIO=4&transport=polling&t=PVxaaZu&sid=ufcpv6MMAw54iD0HAAAC HTTP/1.1" 200 -
WARNING:trading_engine:Session 6 not found in active sessions
INFO:werkzeug:127.0.0.1 - - [11/Jul/2025 15:19:27] "POST /stop_training HTTP/1.1" 200 -
INFO:websocket_handler:Client JEUGqqLUVPBt5TmLAAAD disconnected
INFO:werkzeug:127.0.0.1 - - [11/Jul/2025 15:19:29] "GET /socket.io/?EIO=4&transport=websocket&sid=ufcpv6MMAw54iD0HAAAC HTTP/1.1" 200 -
INFO:websocket_handler:Client T_Qru6Ctf0gBIq1rAAAB disconnected
INFO:websocket_handler:Background broadcast thread stopped
INFO:werkzeug:127.0.0.1 - - [11/Jul/2025 15:19:29] "GET /socket.io/?EIO=4&transport=websocket HTTP/1.1" 500 -
ERROR:werkzeug:Error on request:
Traceback (most recent call last):
  File "/home/alex/miniconda3/envs/ReinforcementTrader/lib/python3.12/site-packages/werkzeug/serving.py", line 370, in run_wsgi
    execute(self.server.app)
  File "/home/alex/miniconda3/envs/ReinforcementTrader/lib/python3.12/site-packages/werkzeug/serving.py", line 336, in execute
    write(b"")
  File "/home/alex/miniconda3/envs/ReinforcementTrader/lib/python3.12/site-packages/werkzeug/serving.py", line 261, in write
    assert status_set is not None, "write() before start_response"
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: write() before start_response
INFO:werkzeug:127.0.0.1 - - [11/Jul/2025 15:19:29] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [11/Jul/2025 15:19:29] "GET /static/css/style.css HTTP/1.1" 304 -
INFO:werkzeug:127.0.0.1 - - [11/Jul/2025 15:19:29] "GET /static/js/main.js HTTP/1.1" 304 -
INFO:werkzeug:127.0.0.1 - - [11/Jul/2025 15:19:29] "GET /sw.js HTTP/1.1" 404 -
INFO:werkzeug:127.0.0.1 - - [11/Jul/2025 15:19:29] "GET /socket.io/?EIO=4&transport=polling&t=PVxadV0 HTTP/1.1" 200 -
INFO:websocket_handler:Client YaTzm4vUQBQrQUldAAAG connected from 127.0.0.1
INFO:werkzeug:127.0.0.1 - - [11/Jul/2025 15:19:29] "POST /socket.io/?EIO=4&transport=polling&t=PVxadV8&sid=OA4HMoSYPJc2XdiZAAAF HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [11/Jul/2025 15:19:29] "GET /socket.io/?EIO=4&transport=polling&t=PVxadV8.0&sid=OA4HMoSYPJc2XdiZAAAF HTTP/1.1" 200 -
INFO:websocket_handler:Client V0qxKMPj_vE8u33OAAAH connected from 127.0.0.1
INFO:werkzeug:127.0.0.1 - - [11/Jul/2025 15:19:29] "GET /socket.io/?EIO=4&transport=polling&t=PVxadVC&sid=OA4HMoSYPJc2XdiZAAAF HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [11/Jul/2025 15:19:29] "GET /socket.io/?EIO=4&transport=polling&t=PVxadVP&sid=OA4HMoSYPJc2XdiZAAAF HTTP/1.1" 200 -
ERROR:data_manager:Error loading data from database: Working outside of application context.

This typically means that you attempted to use functionality that needed
the current application. To solve this, set up an application context
with app.app_context(). See the documentation for more information.
INFO:trading_engine:Training session 7 started
INFO:werkzeug:127.0.0.1 - - [11/Jul/2025 15:19:34] "POST /start_training HTTP/1.1" 200 -
INFO:data_manager:Loading data from data/NQ_data.csv
WARNING:data_manager:Insufficient data in NQ_data.csv: 24 rows
INFO:data_manager:Loading data from data/NQ.csv
INFO:websocket_handler:Client V0qxKMPj_vE8u33OAAAH disconnected
INFO:websocket_handler:Client YaTzm4vUQBQrQUldAAAG disconnected
INFO:websocket_handler:Background broadcast thread stopped
INFO:werkzeug:127.0.0.1 - - [11/Jul/2025 15:19:36] "GET /socket.io/?EIO=4&transport=websocket&sid=OA4HMoSYPJc2XdiZAAAF HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [11/Jul/2025 15:19:36] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [11/Jul/2025 15:19:36] "GET /socket.io/?EIO=4&transport=websocket HTTP/1.1" 500 -
ERROR:werkzeug:Error on request:
Traceback (most recent call last):
  File "/home/alex/miniconda3/envs/ReinforcementTrader/lib/python3.12/site-packages/werkzeug/serving.py", line 370, in run_wsgi
    execute(self.server.app)
  File "/home/alex/miniconda3/envs/ReinforcementTrader/lib/python3.12/site-packages/werkzeug/serving.py", line 336, in execute
    write(b"")
  File "/home/alex/miniconda3/envs/ReinforcementTrader/lib/python3.12/site-packages/werkzeug/serving.py", line 261, in write
    assert status_set is not None, "write() before start_response"
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: write() before start_response
INFO:werkzeug:127.0.0.1 - - [11/Jul/2025 15:19:36] "GET /static/css/style.css HTTP/1.1" 304 -
INFO:werkzeug:127.0.0.1 - - [11/Jul/2025 15:19:36] "GET /static/js/main.js HTTP/1.1" 304 -
INFO:werkzeug:127.0.0.1 - - [11/Jul/2025 15:19:36] "GET /sw.js HTTP/1.1" 404 -
INFO:werkzeug:127.0.0.1 - - [11/Jul/2025 15:19:36] "GET /socket.io/?EIO=4&transport=polling&t=PVxafBi HTTP/1.1" 200 -
INFO:websocket_handler:Client In2sUV8ARDputV8hAAAK connected from 127.0.0.1
INFO:werkzeug:127.0.0.1 - - [11/Jul/2025 15:19:36] "GET /socket.io/?EIO=4&transport=polling&t=PVxafBp.0&sid=UclOSHkuKsG6mGKjAAAJ HTTP/1.1" 200 -
INFO:websocket_handler:Client oSRkujGOKdekto_GAAAL connected from 127.0.0.1
ERROR:data_manager:Error saving data to database: Working outside of application context.

This typically means that you attempted to use functionality that needed
the current application. To solve this, set up an application context
with app.app_context(). See the documentation for more information.
ERROR:data_manager:Error loading NQ data: Working outside of application context.

This typically means that you attempted to use functionality that needed
the current application. To solve this, set up an application context
with app.app_context(). See the documentation for more information.
ERROR:trading_engine:No market data available for session 7
ERROR:trading_engine:Error ending session 7: Working outside of application context.

This typically means that you attempted to use functionality that needed
the current application. To solve this, set up an application context
with app.app_context(). See the documentation for more information.
INFO:websocket_handler:Background broadcast thread started
INFO:werkzeug:127.0.0.1 - - [11/Jul/2025 15:19:36] "POST /socket.io/?EIO=4&transport=polling&t=PVxafBp&sid=UclOSHkuKsG6mGKjAAAJ HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [11/Jul/2025 15:19:36] "GET /socket.io/?EIO=4&transport=polling&t=PVxafBx&sid=UclOSHkuKsG6mGKjAAAJ HTTP/1.1" 200 -